lib/media/drm_engine.js

```diff
@@ -474,6 +475,14 @@ shaka.media.DrmEngine.prototype.prepareMediaKeyConfigs_ =
         variant.drmInfos = [clearKeyDrmInfo];
       }
 
+      // If initial manifest contains unencrypted content,
+      // drm configuration overrides DrmInfo so drmEngine can be activated.
+      // Thus, player can play encrypted content if live stream switches from
+      // unencrypted content to encrypted content during live stream.
+      if (!!configDrmInfos) {
```

Comment:
Only if manifest contains clear content, `configDrmInfos` won't be null. The drawback is that adds drmInfos to each variant although those do not break a playback. I considered the update `configsByKeySystem ` directly instead of modifying manifest, but this way would cause a duplicated code issue. Also, keys are changed for chromecast in this code block. So, I preferred more simple way to add this fix.